[export] Patch dynamo configs when nonstrict tracing

angelayi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #149295

Differential Revision: [D71298929](https://our.internmc.facebook.com/intern/diff/D71298929)

angelayi committed:
[export] Patch dynamo configs when nonstrict tracing

[ghstack-poisoned]

angelayi committed:
Update on "[export] Patch dynamo configs when nonstrict tracing"




[ghstack-poisoned]

angelayi commented:
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D71298929).

angelayi committed:
Update on "[export] Patch dynamo configs when nonstrict tracing"


Differential Revision: [D71298929](https://our.internmc.facebook.com/intern/diff/D71298929)

[ghstack-poisoned]

angelayi committed:
Update on "[export] Patch dynamo configs when nonstrict tracing"


Differential Revision: [D71298929](https://our.internmc.facebook.com/intern/diff/D71298929)

[ghstack-poisoned]

ydwu4 commented on torch/export/_trace.py:
```diff
@@ -116,7 +116,14 @@ class ExportDynamoConfig:
+    automatic_dynamic_shapes: bool = False
+    capture_dynamic_output_shape_ops: bool = True
+    capture_scalar_outputs: bool = True
+    prefer_deferred_runtime_asserts_over_guards: bool = False
```
prefer_deferred_runtime_asserts_over_guards seems to defalut to False in dynamo as well. We may not need it here.

ydwu4 commented on torch/export/_trace.py:
```diff
@@ -1892,7 +1899,17 @@ def _produce_guards_callback(gm):
     tx = TracingContext(fake_mode)
-    with fake_mode, _NonStrictTorchFunctionHandler(), tracing(tx):
+
+    # We also need to attach dynamo configs as these will be used in HOOs that
```
Are these specifically for control flow ops or invoke_subgraph? For control flow ops, we have a _set_compilation_env to put in the flags that're different from the default. 

This PR also changes the behavior of strict export, right?

angelayi replied:
This shouldn't change the behavior of strict export. Those flags are already set in [dynamo export](https://fburl.com/code/5gtic5cg).

These flags directly affect some testing I was doing with invoke_subgraph, but I feel like they would also help with stuff in cond?

angelayi commented:
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D71298929).

angelayi committed:
Update on "[export] Patch dynamo configs when nonstrict tracing"


Differential Revision: [D71298929](https://our.internmc.facebook.com/intern/diff/D71298929)

[ghstack-poisoned]

angelayi commented:
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D71298929).

angelayi committed:
Update on "[export] Patch dynamo configs when nonstrict tracing"


Differential Revision: [D71298929](https://our.internmc.facebook.com/intern/diff/D71298929)

[ghstack-poisoned]

angelayi committed:
Update on "[export] Patch dynamo configs when nonstrict tracing"


Differential Revision: [D71298929](https://our.internmc.facebook.com/intern/diff/D71298929)

[ghstack-poisoned]
